pkg/ds/daemon_set.go

```diff
@@ -158,14 +163,23 @@ func (ds *daemonSet) WatchDesires(
 			if err != nil {
 				select {
 				case errCh <- err:
-					// Retry the replication for extra robustness
-					time.Sleep(RetryInterval)
-					err = ds.PublishToReplication()
+					ds.logger.Infof("An error has occurred in the daemon set, retrying if no changes are made in %v", RetryInterval)
+					// Retry the replication in the RetryInterval's duration
+					timer.Reset(RetryInterval)
+					// This is required in case the user disables the daemon set
+					// so that the timer would be stopped after
+					err = nil
```

Comment:
Yeah, it gets logged by the farm, the `Infof` a few lines above are a little more verbose in saying a retry will occur if nothing else happens.